neuror/sanitize.py

```diff
@@ -37,7 +45,16 @@ def sanitize(input_neuron, output_path):
     if neuron.soma.type == SomaType.SOMA_UNDEFINED:  # pylint: disable=no-member
         raise CorruptedMorphology('{} has no soma'.format(input_neuron))
     if np.any(neuron.diameters < 0):
-        raise CorruptedMorphology('{} negative diameters'.format(input_neuron))
+        raise CorruptedMorphology('{} has negative diameters'.format(input_neuron))
+
+    for root in neuron.root_sections:  # pylint: disable=not-an-iterable
+        for section in root.iter():
+            if section.type != root.type:
+                raise CorruptedMorphology(f'{input_neuron} has a neurite whose type changes along '
+                                          'the way\n'
+                                          f'Child section (id: {section.id}) has a different type '
+                                          f'({section.type}) than its parent (id: '
```

Comment:
The message is not consistent with what is actually checked (the current section is compared to the root, not to the parent).
The `root.iter()` generator should ensure that the parent always has the same type as the root, but still, it is better to be consistent (maybe one day the `root.iter()` will change and this kind of inconsistencies might make it harder to debug).

Reply:
@wizmer  should know that, this file is not supposed to be updated in this PR, if this is a real by,  maybe  we should start a new PR?

Reply:
Yes, good eyes Adrien, the message is not consistent but as you said it works because the root section and the parent will always be the same. We could change it to compare to the parent for real, yes.